upgrade/steps.go

```diff
@@ -841,24 +834,19 @@ func AddAutoAliasing(ctx Context, repo ProviderRepo, providerName string) (step.
 	return step.Combined("Add AutoAliasing", steps...), nil
 }
 
-func ReplaceAssertNoError(ctx Context, repo ProviderRepo, providerName string) (step.Step, error) {
+func AddAutoAliasing(ctx Context, repo ProviderRepo, providerName string) (step.Step, error) {
 	steps := []step.Step{}
-	changesMade := false
-	steps = append(steps,
-		step.F("Replace AssertNoError", func() (string, error) {
-			changes, err := AssertNoErrorMigration(filepath.Join(*repo.providerDir(), "resources.go"))
-			if err != nil {
-				return "failed to perform AssertNoError migration", err
-			}
-			changesMade = changes
-			return "", err
-		}))
-	if changesMade {
-		steps = append(steps,
-			step.Cmd(exec.CommandContext(ctx, "gofmt", "-s", "-w", "resources.go")).In(repo.providerDir()),
-			step.Cmd(exec.CommandContext(ctx, "git", "add", "resources.go")).In(&repo.root),
-			step.Cmd(exec.CommandContext(ctx, "git", "commit", "-m", "assertnoerror migration")).In(&repo.root),
-		)
+	metadataPath := fmt.Sprintf("%s/cmd/pulumi-resource-%s/bridge-metadata.json", *repo.providerDir(), providerName)
+	steps = append(steps, step.Cmd(exec.CommandContext(ctx, "git", "add", metadataPath)).In(&repo.root))
+	migrationSteps, err := migrationSteps(ctx, repo, providerName, "Add AutoAliasing", AutoAliasingMigration)
+	if err != nil {
+		return nil, err
 	}
-	return step.Combined("Replace AssertNoError", steps...), nil
+	steps = append(steps, migrationSteps)
+	return step.Combined("Add AutoAliasing", steps...), nil
+}
+
+func ReplaceAssertNoError(ctx Context, repo ProviderRepo, providerName string) (step.Step, error) {
```

Comment:
`providerName` is unused here, but allows `migrationSteps()` to take in `migrationFunc func(resourcesFilePath, providerName string) (bool, error)` as an arg

Reply:
It looks like you pass `provideName` to `migrationSteps` here.

Either way, this kind of clarifying comment is great as a comment in your code.